Content.Server/Power/Generator/SolidFuelGeneratorAdapterComponent.cs

```diff
@@ -1,23 +1,40 @@
 ﻿using Content.Shared.Materials;
+using Content.Shared.Power.Generator;
 using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
 
 namespace Content.Server.Power.Generator;
 
 /// <summary>
-/// This is used for allowing you to insert fuel into gens.
+/// Fuels a <see cref="FuelGeneratorComponent"/> through solid materials.
 /// </summary>
+/// <remarks>
+/// <para>
+/// Must be accompanied with a <see cref="MaterialStorageComponent"/> to store the actual material and handle insertion logic.
+/// You should set a whitelist there for the fuel material.
+/// </para>
+/// <para>
+/// The component itself stores a "fractional" fuel value to allow stack materials to be gradually consumed.
+/// </para>
+/// </remarks>
 [RegisterComponent, Access(typeof(GeneratorSystem))]
 public sealed class SolidFuelGeneratorAdapterComponent : Component
 {
     /// <summary>
     /// The material to accept as fuel.
     /// </summary>
-    [DataField("fuelMaterial", customTypeSerializer: typeof(PrototypeIdSerializer<MaterialPrototype>)), ViewVariables(VVAccess.ReadWrite)]
+    [DataField("fuelMaterial", customTypeSerializer: typeof(PrototypeIdSerializer<MaterialPrototype>))]
+    [ViewVariables(VVAccess.ReadWrite)]
     public string FuelMaterial = "Plasma";
 
     /// <summary>
-    /// How much fuel that material should count for.
+    /// How much material (can be fractional) is left in the generator.
+    /// </summary>
+    [DataField("fractionalMaterial"), ViewVariables(VVAccess.ReadWrite)]
+    public float FractionalMaterial;
```

Comment:
not sure i understand the necessity of these `fractional*` variables? on this & the chem adapter

Reply:
With the previous generator code, dumping fuel/sheets into the generator would just delete them and add to a fuel counter. Now with the changes in the PR we actually store the real reagents instead.

The issue though is that the fuel consumption values are *way* smaller than the epsilon of both reagents and material counts. If we just subtracted 0.01 welding fuel or 1cm3 of plasma every tick, then the slowest you can burn fuel is having a single unit or sheet last at most a second and a half. That's not very practical.

So this allows us to further subdivide the "input" material/reagent counts.

Reply:
aight